dashboard/app/views/levels/show.html.haml

```diff
@@ -26,8 +26,11 @@
     if (appOptions.autoplayVideo) {
       showVideoDialog(appOptions.autoplayVideo);
     }
-  = render partial: "levels/#{@level.class.to_s.underscore}"
-  = render partial: 'levels/reference_area'
+  - if @level.is_a?(Widget)
+    = render partial: 'levels/widget'
+    = render partial: 'levels/reference_area'
+  - else
+    = render partial: "levels/#{@level.class.to_s.underscore}"
```

Comment:
Not something to do in this PR, but maybe as you continue refactoring:

This whole big conditional looks like it may want to turn into just
````
= render partial: "levels/#{@level.class.to_s.underscore}"
```` 

which would kind of better match the way inheritance tree is set up in the model class structure (ie. the caller of a method on an instance of a Pixelation is calling methods on a Pixelation -- internally the methods are on Level or Widget (or can be overriden, etc) but the caller doesn't really care). So `levels/_pixelation` might just call `levels/_widget` and `levels/_all`, or it might call `levels_widget` and then do some of its own stuff, etc.

It may not fit exactly (like you don't actually have methods and super in views) but it may make things like this (eg. figuring out who is rendering reference_area) easier to follow.